StockPriceScraper/Program.cs

```diff
@@ -150,23 +150,34 @@ private static StockData DownloadWebsite(string url)
             var stockPriceEurNode = htmlDoc.DocumentNode.SelectSingleNode(eurXpath);
             var stockPriceUsdNode = htmlDoc.DocumentNode.SelectSingleNode(usdXpath);
 
-            double stockPriceUsd = 0.0;
-            double stockPriceEur = 0.0;
+            decimal stockPriceUsd = default;
+            decimal stockPriceEur = default;
 
             if (stockPriceEurNode.InnerHtml.Contains("EUR"))
             {
-                stockPriceEur = Double.Parse(stockPriceEurNode.FirstChild.InnerText);
+                stockPriceEur = ParseGermanDecimal(stockPriceEurNode.FirstChild.InnerText);
```

Comment:
Decimal vs double? I know decimal has a higher precision, but this is not needed here. 
I do understand the parsing, didn't know that there is an extra method for it.

Reply:
I created the ParseGermanDecimal function so that it can be reused as we need it twice (for EUR and USD). Later then this becomes an extension method which mainly supports readbility. [https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/extension-methods](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/extension-methods)